### PR TITLE
[#982] Implement isCurrentScene convenience method

### LINKED ISSUE #982: Add convenience method to determine if a given Scene is the current scene
### Context
`Engine.currentScene` returns the current scene, but there is not a way from within `Scene` itself to determine if that scene is the current scene. You can achieve this by checking `engine.currentScene === scene`, but this requires that you take a dependency on `engine` and we should make this more accessible.

### Proposal
Add an `isCurrentScene` method to `Scene`. We can track this state by hooking into the `onActivate` and `onDeactivate` events. This would be useful in circumstances where an event is bound to the engine but we only want its callback to execute in a specific scene, like we did [here](https://github.com/excaliburjs/ludum-41/blob/1549e06a953ead1e0a56fb1d931ba5c64eaa4303/game/top/top.ts#L137).

### PR DESCRIPTION
**Pull request by FerociousQuasar**
Closes #982 

## Changes:

- added isCurrentScene() to scene objects
..* This is property is updated on activate(), deactivate() which is managed by engine.gotoScene.

## Note:
Have not created/run unit tests.  This is forthcoming.

### CODE REVIEW

File: src/engine/Scene.ts [hunk 1]
```diff
@@ -223,7 +224,7 @@ export class Scene extends Class implements ICanInitialize, ICanActivate, ICanDe
          this._initializeChildren();
          
          this._logger.debug('Scene.onInitialize', this, engine);
-
+
```
Comment (eonarheim) [inline src/engine/Scene.ts]: Let's remove this added whitespace

File: src/engine/Scene.ts [hunk 2]
```diff
@@ -732,6 +734,12 @@ export class Scene extends Class implements ICanInitialize, ICanActivate, ICanDe
       this._sortedDrawingTree.add(actor);
    }
 
+   public isCurrentScene():boolean {
```
Comment (eonarheim) [inline src/engine/Scene.ts]: Fix linting error, space between `:` and `boolean`

**Comment by alanag13** [comment]
That's a very valid point. Right now engine is only passed to scene's ctor optionally, so  i would be more in favor of the setter you mentioned rather than asking `engine.currentScene == this`, otherwise it would mean a hard dependency on engine.

**Comment by eonarheim** [comment]
@alanag13 I don't think it's an issue, scenes that have yet to be associated with the engine cannot be the active scene.

**Comment by eonarheim** [comment]
Also scenes cannot run without engine, so they already have a hard dependency.

File: src/engine/Scene.ts [hunk 3]
```diff
@@ -734,12 +731,13 @@ export class Scene extends Class implements ICanInitialize, ICanActivate, ICanDe
       this._sortedDrawingTree.add(actor);
    }
 
-   public isCurrentScene():boolean {
-      return this._isCurrentScene;
+   public isCurrentScene(): boolean {
```
Comment (kamranayub) [inline src/engine/Scene.ts]: Thanks, this feels much better

**Pull request merged**